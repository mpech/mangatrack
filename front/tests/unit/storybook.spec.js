import registerRequireContextHook from 'babel-plugin-require-context-hook/register'
import initStoryshots from '@storybook/addon-storyshots'
jest.mock('moment', () => () => ({ fromNow: () => '2 days ago', format: () => '3 days ago' }))
registerRequireContextHook()
initStoryshots()
