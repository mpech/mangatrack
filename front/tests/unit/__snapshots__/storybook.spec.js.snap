// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Batch at, link, status 1`] = `
<table
  class="pure-table"
>
  <tr>
    <td>
      2 days ago
    </td>
    <td>
      https://mangakakalot.com/manga/versatile_mage
    </td>
    <td>
      OK
    </td>
  </tr>
</table>
`;

exports[`Storyshots Batch link, status, at 1`] = `
<table
  class="pure-table"
>
  <tr>
    <td>
      https://mangakakalot.com/manga/versatile_mage
    </td>
    <td>
      OK
    </td>
    <td>
      2 days ago
    </td>
  </tr>
</table>
`;

exports[`Storyshots BatchList default 1`] = `
<table
  class="pure-table batchList"
>
  <thead>
    <tr>
      <th>
        status
      </th>
       
      <th>
        link
      </th>
       
      <th>
        at
      </th>
    </tr>
  </thead>
   
  <tbody />
   
  <tfoot>
    <tr>
      <td
        colspan="3"
      >
        <ul
          class="pagination"
          data-v-82963a40=""
        >
          <!---->
           
          <li
            class="disabled"
            data-v-82963a40=""
          >
            <a
              data-v-82963a40=""
              tabindex="-1"
            >
              Prev
            </a>
          </li>
            
          <li
            class="disabled"
            data-v-82963a40=""
          >
            <a
              data-v-82963a40=""
              tabindex="-1"
            >
              Next
            </a>
          </li>
           
          <!---->
        </ul>
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`Storyshots BatchList two per page 1`] = `
<table
  class="pure-table batchList"
>
  <thead>
    <tr>
      <th>
        status
      </th>
       
      <th>
        link
      </th>
       
      <th>
        at
      </th>
    </tr>
  </thead>
   
  <tbody />
   
  <tfoot>
    <tr>
      <td
        colspan="3"
      >
        <ul
          class="pagination"
          data-v-82963a40=""
        >
          <!---->
           
          <li
            class="disabled"
            data-v-82963a40=""
          >
            <a
              data-v-82963a40=""
              tabindex="-1"
            >
              Prev
            </a>
          </li>
            
          <li
            class="disabled"
            data-v-82963a40=""
          >
            <a
              data-v-82963a40=""
              tabindex="-1"
            >
              Next
            </a>
          </li>
           
          <!---->
        </ul>
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`Storyshots BatchPoller should not notify (pending) 1`] = `<div />`;

exports[`Storyshots BatchPoller should not notify shortDelayed 1`] = `<div />`;

exports[`Storyshots BatchPoller should notify OK 1`] = `<div />`;

exports[`Storyshots Card default 1`] = `
<div
  class="pure-u-1-2 pure-u-lg-1-6"
>
  <div
    class="card"
  >
    <a
      class=""
      href="#/xx/nid"
    >
      <figure>
        <img
          src="https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png"
        />
      </figure>
    </a>
     
    <div>
      <h4
        title="some name"
      >
        <span>
          <a
            class=""
            href="#/xx/nid"
          >
            some name
          </a>
        </span>
      </h4>
       
      <div
        class="chapter"
      >
        <span>
          <a
            href="lastchapurl"
          >
            c
            <span>
              2
            </span>
            ↗
          </a>
        </span>
         
        <time
          updatedat="1577206158382"
        >
          2 days ago
        </time>
      </div>
    </div>
     
    <hr />
     
    <div
      style="position: relative;"
    >
      <!---->
       
      <span>
        <span
          class="mt-follow"
          title="Track some name"
        >
          <a
            href="#"
          >
            ♥
          </a>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Follow default 1`] = `
<span
  title="Test"
>
  <span
    class="mt-follow"
    title="Track def"
  >
    <a
      href="#"
    >
      ♥
    </a>
  </span>
</span>
`;

exports[`Storyshots Follow followed 1`] = `
<span
  title="Test"
>
  <span
    class="followed mt-follow"
    title="Untrack def"
  >
    <a
      href="#"
    >
      ♥
    </a>
  </span>
</span>
`;

exports[`Storyshots Grid default 1`] = `
<div
  class="mt-grid"
>
  <div
    class="pure-g"
  >
    <div
      class="pure-u-1-2 pure-u-lg-1-6"
    >
      <div
        class="card"
      >
        <a
          class=""
          href="#/manga/nid"
        >
          <figure>
            <img
              src="https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png"
            />
          </figure>
        </a>
         
        <div>
          <h4
            title="some name"
          >
            <span>
              <a
                class=""
                href="#/manga/nid"
              >
                some name
              </a>
            </span>
          </h4>
           
          <div
            class="chapter"
          >
            <span>
              <a
                href="lastchapurl"
              >
                c
                <span>
                  2
                </span>
                ↗
              </a>
            </span>
             
            <time
              updatedat="1577206158382"
            >
              2 days ago
            </time>
          </div>
        </div>
         
        <hr />
         
        <div
          style="position: relative;"
        >
          <span
            class="card-stats"
          >
            1/2
          </span>
           
          <span>
            <span
              class="followed mt-follow"
              title="Untrack some name"
            >
              <a
                href="#"
              >
                ♥
              </a>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="pure-u-1-2 pure-u-lg-1-6"
    >
      <div
        class="card"
      >
        <a
          class=""
          href="#/manga/nid"
        >
          <figure>
            <img
              src="https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png"
            />
          </figure>
        </a>
         
        <div>
          <h4
            title="other"
          >
            <span>
              <a
                class=""
                href="#/manga/nid"
              >
                other
              </a>
            </span>
          </h4>
           
          <div
            class="chapter"
          >
            <span>
              <a
                href="lastchapurl"
              >
                c
                <span>
                  2
                </span>
                ↗
              </a>
            </span>
             
            <time
              updatedat="1577206158382"
            >
              2 days ago
            </time>
          </div>
        </div>
         
        <hr />
         
        <div
          style="position: relative;"
        >
          <!---->
           
          <span>
            <span
              class="mt-follow"
              title="Track other"
            >
              <a
                href="#"
              >
                ♥
              </a>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="mt-pagination"
  >
    <a
      class="pure-button"
      href="#"
    >
      Moar
    </a>
  </div>
</div>
`;

exports[`Storyshots Grid no pagination (deactivated) 1`] = `
<div
  class="mt-grid"
>
  <div
    class="pure-g"
  >
    <div
      class="pure-u-1-2 pure-u-lg-1-6"
    >
      <div
        class="card"
      >
        <a
          class=""
          href="#/manga/nid"
        >
          <figure>
            <img
              src="https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png"
            />
          </figure>
        </a>
         
        <div>
          <h4
            title="some name"
          >
            <span>
              <a
                class=""
                href="#/manga/nid"
              >
                some name
              </a>
            </span>
          </h4>
           
          <div
            class="chapter"
          >
            <span>
              <a
                href="lastchapurl"
              >
                c
                <span>
                  2
                </span>
                ↗
              </a>
            </span>
             
            <time
              updatedat="1577206158382"
            >
              2 days ago
            </time>
          </div>
        </div>
         
        <hr />
         
        <div
          style="position: relative;"
        >
          <span
            class="card-stats"
          >
            1/2
          </span>
           
          <span>
            <span
              class="followed mt-follow"
              title="Untrack some name"
            >
              <a
                href="#"
              >
                ♥
              </a>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="pure-u-1-2 pure-u-lg-1-6"
    >
      <div
        class="card"
      >
        <a
          class=""
          href="#/manga/nid"
        >
          <figure>
            <img
              src="https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png"
            />
          </figure>
        </a>
         
        <div>
          <h4
            title="other"
          >
            <span>
              <a
                class=""
                href="#/manga/nid"
              >
                other
              </a>
            </span>
          </h4>
           
          <div
            class="chapter"
          >
            <span>
              <a
                href="lastchapurl"
              >
                c
                <span>
                  2
                </span>
                ↗
              </a>
            </span>
             
            <time
              updatedat="1577206158382"
            >
              2 days ago
            </time>
          </div>
        </div>
         
        <hr />
         
        <div
          style="position: relative;"
        >
          <!---->
           
          <span>
            <span
              class="mt-follow"
              title="Track other"
            >
              <a
                href="#"
              >
                ♥
              </a>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`Storyshots Grid no pagination (no more) 1`] = `
<div
  class="mt-grid"
>
  <div
    class="pure-g"
  >
    <div
      class="pure-u-1-2 pure-u-lg-1-6"
    >
      <div
        class="card"
      >
        <a
          class=""
          href="#/manga/nid"
        >
          <figure>
            <img
              src="https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png"
            />
          </figure>
        </a>
         
        <div>
          <h4
            title="some name"
          >
            <span>
              <a
                class=""
                href="#/manga/nid"
              >
                some name
              </a>
            </span>
          </h4>
           
          <div
            class="chapter"
          >
            <span>
              <a
                href="lastchapurl"
              >
                c
                <span>
                  2
                </span>
                ↗
              </a>
            </span>
             
            <time
              updatedat="1577206158382"
            >
              2 days ago
            </time>
          </div>
        </div>
         
        <hr />
         
        <div
          style="position: relative;"
        >
          <!---->
           
          <span>
            <span
              class="mt-follow"
              title="Track some name"
            >
              <a
                href="#"
              >
                ♥
              </a>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="pure-u-1-2 pure-u-lg-1-6"
    >
      <div
        class="card"
      >
        <a
          class=""
          href="#/manga/nid"
        >
          <figure>
            <img
              src="https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png"
            />
          </figure>
        </a>
         
        <div>
          <h4
            title="other"
          >
            <span>
              <a
                class=""
                href="#/manga/nid"
              >
                other
              </a>
            </span>
          </h4>
           
          <div
            class="chapter"
          >
            <span>
              <a
                href="lastchapurl"
              >
                c
                <span>
                  2
                </span>
                ↗
              </a>
            </span>
             
            <time
              updatedat="1577206158382"
            >
              2 days ago
            </time>
          </div>
        </div>
         
        <hr />
         
        <div
          style="position: relative;"
        >
          <!---->
           
          <span>
            <span
              class="mt-follow"
              title="Track other"
            >
              <a
                href="#"
              >
                ♥
              </a>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`Storyshots LinkImporter default 1`] = `
<form
  action="#incase"
  class="pure-form linkImporter"
>
  <fieldset>
    <legend>
      Import from a link
    </legend>
     
    <div>
      <table
        class="mq-table pure-table-bordered pure-table"
      >
        <caption>
          Examples
        </caption>
         
        <thead>
          <th>
            from
          </th>
           
          <th>
            url
          </th>
        </thead>
         
        <tbody>
          <tr>
            <td>
              mangakakalot
            </td>
            <td>
              https://mangakakalot.com/manga/versatile_mage
            </td>
          </tr>
           
          <tr>
            <td>
              fanfox
            </td>
            <td>
              https://fanfox.net/manga/versatile_mage/
            </td>
          </tr>
        </tbody>
      </table>
    </div>
     
    <div
      class="fields"
    >
      <input
        class="pure-input-1-2"
        pattern="^https://(mangakakalot.com|manganelo.com|fanfox.net)/manga/[^/]+/?"
        placeholder="https://mangakakalot.com/manga/versatile_mage"
        required="required"
        type="text"
      />
       
      <button
        class="pure-button pure-button-primary pure-input"
        type="submit"
      >
        Import
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`Storyshots MangaView default 1`] = `
<div
  class="mangaView"
>
  <h1>
    
  </h1>
   
  <div
    class="pure-g"
  >
    <div
      class="pure-u-1-5"
    >
      <figure>
        <img />
      </figure>
    </div>
     
    <div
      class="pure-u-4-5"
    >
      <!---->
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`Storyshots MangaView html escaped but htmlentities decoded 1`] = `
<div
  class="mangaView"
>
  <h1>
    
  </h1>
   
  <div
    class="pure-g"
  >
    <div
      class="pure-u-1-5"
    >
      <figure>
        <img />
      </figure>
    </div>
     
    <div
      class="pure-u-4-5"
    >
      <!---->
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`Storyshots MangaView without description 1`] = `
<div
  class="mangaView"
>
  <h1>
    
  </h1>
   
  <div
    class="pure-g"
  >
    <div
      class="pure-u-1-5"
    >
      <figure>
        <img />
      </figure>
    </div>
     
    <div
      class="pure-u-4-5"
    >
      <!---->
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`Storyshots Menu logged admin 1`] = `
<div
  class="pure-menu pure-menu-horizontal"
>
  <a
    class="pure-menu-heading pure-menu-link"
    href="/"
  >
    MangaTrack
  </a>
   
  <a
    class="pure-menu-heading pure-menu-link"
    href="/me"
    title="my space"
  >
    Me
  </a>
   
  <a
    class="pure-menu-heading pure-menu-link login hide"
    href="#/login"
    title="Mainly sync your tracked stuff"
  >
    Login
  </a>
   
  <a
    class="pure-menu-heading pure-menu-link logout"
    href="#/login"
  >
    Logout
  </a>
   
  <a
    class="pure-menu-heading pure-menu-link admin"
    href="#/admin"
  >
    Admin
  </a>
</div>
`;

exports[`Storyshots Menu no logged no admin 1`] = `
<div
  class="pure-menu pure-menu-horizontal"
>
  <a
    class="pure-menu-heading pure-menu-link"
    href="/"
  >
    MangaTrack
  </a>
   
  <a
    class="pure-menu-heading pure-menu-link"
    href="/me"
    title="my space"
  >
    Me
  </a>
   
  <a
    class="pure-menu-heading pure-menu-link login"
    href="#/login"
    title="Mainly sync your tracked stuff"
  >
    Login
  </a>
   
  <a
    class="pure-menu-heading pure-menu-link logout hide"
    href="#/login"
  >
    Logout
  </a>
   
  <a
    class="pure-menu-heading pure-menu-link admin hide"
    href="#/admin"
  >
    Admin
  </a>
</div>
`;

exports[`Storyshots Pagination default 1`] = `
<div
  class="mt-pagination"
>
  <a
    class="pure-button"
    href="#"
  >
    Moar
  </a>
</div>
`;
